wordpress-mobile/WordPress-iOS
# Move Shared code into WPiOS as a dynamic framework

Issue #7139 
(this PR tries to be as similar as possible as: https://github.com/wordpress-mobile/WordPress-iOS/pull/7121)

This PR brings `WordPressShared` into the main project as a dynamic framework.

* A new project was created called `WordPressShared`. This project has two targets - a dynamic framework and a test target. All of the code and configuration settings were made to mirror what the original `Shared` project podspec with the exception of some minor reorganization of the files and folder renaming.
* `WordPressShared` requires CocoaPods for logging. The Podfile was updated to support this new project and a shared pod definitions was placed into a macro of sorts. The `WordPressShared` project is in its own root-level target.
* Manually configured the main target to embed the framework. 
* Added the framework to the linker for the Today and Share widgets.
* Added the stats unit tests to the scheme for unit tests in the main target.

## To Test
1. Update your local CocoaPods repo with `pod repo update`.
2. Clean your `DerivedData` folder.
3. Build and test main WordPress target. All tests should pass.
4. Run main WordPress target.
5. Go to the reader, check that the Noto font has loaded.
6. Explore all the app, stats particularly, and check that everything looks as it used to. There should be no visible changes.
7. Switch to the WordPressShared scheme and run unit tests.
8. Keep playing around, nothing should break.

Needs review: @astralbodies, @nheagy, @koke, @aerych 

With special thanks to @astralbodies for all the guidance during this one.
